src/shared/src/types.rs

```diff
@@ -100,3 +100,57 @@ pub mod custom_token {
         Icrc(LedgerId),
     }
 }
+
+/// Types specifics to the user profile.
+pub mod user_profile {
+    use candid::{CandidType, Deserialize, Principal};
+    use std::collections::BTreeMap;
+
+    pub type CredentialType = String;
+
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct UserCredential {
+        pub credential_type: CredentialType,
+        pub verified_date_timestamp: Option<u64>,
+        pub expire_date_timestamp: Option<u64>,
+    }
+
+    // Used in the endpoint
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct UserProfile {
+        pub credentials: Vec<UserCredential>,
+        pub created_timestamp: u64,
+        pub updated_timestamp: u64,
+    }
+
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct UserProfileStorage {
+        pub credentials: BTreeMap<CredentialType, UserCredential>,
+        pub created_timestamp: u64,
+        pub updated_timestamp: u64,
+    }
+
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct AddCredentialRequest {
+        pub credential_jwt: String,
+    }
+
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct GetUsersRequest {
+        pub updated_after_timestamp: Option<u64>,
+        pub limit_response: Option<u64>,
+    }
+
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct OisyUser {
```

Comment:
Why not just `User` out of curiosity?

Reply:
It was even called `PrivateUser`. This is more external and I believe `User` seems like an internal type instead.

Reply:
I see. I just feld "OisyUser" not super meaningful but, does not have a better naming to suggest.

Reply:
Naming is hard...